test/alternate_entrypoint/altentry_test.go

```diff
@@ -0,0 +1,56 @@
+package alternate_entrypoint
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestValidEntrypoints(t *testing.T) {
+	cases := []struct {
+		in         string
+		entrypoint string
+	}{
+		{"aacc", ""},
+		{"bbbcc", "Entry2"},
+		{"cc", "Entry3"},
+		{"cc", "C"},
+	}
+
+	for _, c := range cases {
+		v, err := Parse("", []byte(c.in), Entrypoint(c.entrypoint))
+		if err != nil {
+			t.Errorf("%s:%s: got error %s", c.entrypoint, c.in, err)
+		}
+
+		got := string(v.([]byte))
+		if got != c.in {
+			t.Errorf("%s:%s: got %s", c.entrypoint, c.in, got)
+		}
+	}
+}
+
+func TestInvalidEntrypoints(t *testing.T) {
+	cases := []struct {
+		in         string
+		entrypoint string
+		errMsg     string
+	}{
+		{"bbbcc", "Z", errInvalidEntrypoint.Error()},
```

Comment:
Maybe one of the tests should reference rule `A` or `B`, which are optimized away (not that important).